### PR TITLE
Research and Development Fixes

### PR DESCRIPTION
**Pull request by FarmWizard**
## Pre-Merge Checklist
- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

image
image
image
image
image
image
image
image
image
image
image
image
image
Here's all the stuff you can make with the protolathe after my changes. Ran on a private instance.

As for the surgery changes. The only thing added is a pacification surgery. It's a very late in the techtree surgery that lets you give someone the 'pacifist' trait essentially. I don't think the potential for roundgrief is very high. Infact it'll probably just be used against 'le funny raider' types by nash since you have to do lategame research and then plug a special disk into the surgery console and then finally get whoever you're doing it on at that surgery table and actually do it.

As for the stock parts. I reverted those changes. 
## Changelog
:cl:
add: sciense
/:cl:

### CODE REVIEW

File: code/modules/research/techweb/nodes/tools_nodes.dm [hunk 1]
```diff
@@ -13,15 +13,15 @@
 	display_name = "Mining Technology"
 	description = "Better than Efficiency V."
 	prereq_ids = list("engineering", "basic_plasma")
-	design_ids = list("drill", "superresonator", "triggermod", "damagemod", "cooldownmod", "rangemod", "ore_redemption", "mining_equipment_vendor", "cargoexpress")//CB Edit Yeet Plasma cutter
+	design_ids = list("drill", "superresonator", "triggermod", "damagemod", "cooldownmod", "rangemod", "ore_redemption", "mining_equipment_vendor", "cargoexpress", "plasmacutter")//e a r l y    g a  m e)
```
Comment (GremlingSS) [inline code/modules/research/techweb/nodes/tools_nodes.dm:16]: Unsure if the plasma cutter was nintended to be readded? but eh, seems good, idk what implications it has.

File: code/modules/research/designs.dm [hunk 2]
```diff
@@ -45,7 +45,6 @@ other types of metals and chemistry for reagents).
 	/// Checks if the material price is less than that of what it gives back, and then adds some if so
 	/// Spams the chat with bullshit if so, its a debugging tool~
 	var/debug_materials = FALSE
-
```
Comment (GremlingSS) [inline code/modules/research/designs.dm:48]: pain

**Review by GremlingSS — APPROVED** [review]
I approve, some weird bits of code which doesn't impact much, but my OCD kills.

There is a concern over the plasma cutter, but idk what this means in terms of gameplay and why we removed it.

**Comment by GremlingSS** [comment]
Just got told the plasma cutter was removed cause we suspected it to cause server crashes (which it wasn't, it was just auxmos being auxmos lol)

File: code/modules/research/designs/misc_designs.dm [hunk 3]
```diff
@@ -575,7 +575,7 @@
 	id = "knight_helmet"
 	build_type = AUTOLATHE
 	materials = list(MAT_CATEGORY_RIGID = 5000)
-	build_path = /obj/item/clothing/head/helmet/knight/greyscale
+	build_path = /obj/item/clothing/head/helmet/knight/greyscale
```
[the remark this answers is not in the document]
Comment (FarmWizard) [reply code/modules/research/designs/misc_designs.dm:578]: Fixed

File: code/modules/research/designs/weapon_designs.dm [hunk 4]
```diff
@@ -504,7 +504,7 @@
 	build_type = AUTOLATHE
 	materials = list(MAT_CATEGORY_RIGID = 12000)
 	build_path = /obj/item/melee/cleric_mace
-	category = list("Imported")
+	category = list("Imported")
```
[the remark this answers is not in the document]
Comment (FarmWizard) [reply code/modules/research/designs/weapon_designs.dm:507]: Fixed

File: code/modules/research/techweb/nodes/tools_nodes.dm (hunk 1 above)
Comment (FarmWizard) [reply code/modules/research/techweb/nodes/tools_nodes.dm:16]: Then it's guuchi?

File: code/modules/research/designs.dm (hunk 2 above)
Comment (FarmWizard) [reply code/modules/research/designs.dm:48]: fixed

File: code/modules/research/stock_parts.dm
Comment (GremlingSS) [inline code/modules/research/stock_parts.dm:1]: Fenny and Dan doesn't like that you removed these stock parts, I don't understand why they were removed either, and why you reset the other stockpart files back to the originals.

**Review by GremlingSS — CHANGES_REQUESTED** [review]

File: code/modules/research/stock_parts.dm
Comment (GremlingSS) [reply code/modules/research/stock_parts.dm:1]: without setting the originals to be them instead, I just don't understand your thought process on that either.

File: code/game/objects/effects/spawners/lootdrop.dm [hunk 5]
```diff
@@ -667,11 +667,11 @@
...
-				/obj/item/stock_parts/capacitor/simple,
-				/obj/item/stock_parts/scanning_module/simple,
-				/obj/item/stock_parts/manipulator/simple,
-				/obj/item/stock_parts/micro_laser/simple,
-				/obj/item/stock_parts/matter_bin/simple,
+				/obj/item/stock_parts/capacitor,
+				/obj/item/stock_parts/scanning_module,
+				/obj/item/stock_parts/manipulator,
+				/obj/item/stock_parts/micro_laser,
+				/obj/item/stock_parts/matter_bin,
```
Comment (GremlingSS) [inline code/game/objects/effects/spawners/lootdrop.dm:674]: this is also in regards to the stockpart issue.

File: code/modules/research/stock_parts.dm
Comment (FarmWizard) [reply code/modules/research/stock_parts.dm:1]: reverted

**Comment by FarmWizard** [comment]
The integration failure is a result of step_trigger which is code I haven't touched.

File: code/game/objects/effects/step_triggers.dm [hunk 6]
```diff
@@ -4,6 +4,7 @@
 	var/affect_ghosts = 0
 	var/stopper = 1 // stops throwers
 	var/mobs_only = FALSE
+	var/client = 0 // I don't know what the heck is pulling var/client, but this shit runtimes if it can't access it. Don't use this trigger until it's fixed
```
Comment (GremlingSS) [inline code/game/objects/effects/step_triggers.dm:7]: Revert this. I'll fix it.

File: code/game/objects/items/devices/portable_chem_blenderbelt.dm [hunk 7]
```diff
@@ -796,8 +796,8 @@
 	new /obj/item/stock_parts/cell/upgraded(src)
 	new /obj/item/storage/belt/military/snack(src)
 	new /obj/item/storage/belt/military/snack(src)
-	new /obj/item/stock_parts/manipulator/simple(src)
-	new /obj/item/stock_parts/manipulator/simple(src)
+	new /obj/item/stock_parts/manipulator(src)
+	new /obj/item/stock_parts/manipulator(src)
```
Comment (GremlingSS) [inline code/game/objects/items/devices/portable_chem_blenderbelt.dm:800]: Pls revert this too

File: code/modules/research/designs/medical_designs.dm [hunk 8]
```diff
@@ -617,7 +616,7 @@
 	name = "Cybernetic Liver"
 	desc = "A cybernetic liver"
 	id = "cybernetic_liver"
-	build_type = MECHFAB
+	build_type = PROTOLATHE
```
Comment (GremlingSS) [inline code/modules/research/designs/medical_designs.dm:619]: Add `| MECHFAB`

File: code/modules/research/designs/medical_designs.dm [hunk 9]
```diff
@@ -627,7 +626,7 @@
 	name = "Cybernetic Heart"
 	desc = "A cybernetic heart"
 	id = "cybernetic_heart"
-	build_type = MECHFAB
+	build_type = PROTOLATHE
 	materials = list(/datum/material/iron = 500, /datum/material/glass = 500)
```
Comment (GremlingSS) [inline code/modules/research/designs/medical_designs.dm:630]: Add `| MECHFAB`

File: code/modules/research/designs/medical_designs.dm [hunk 10]
```diff
@@ -648,7 +647,7 @@
 	name = "Upgraded Cybernetic Liver"
 	desc = "An upgraded cybernetic liver"
 	id = "cybernetic_liver_u"
-	build_type = MECHFAB
+	build_type = PROTOLATHE
```
Comment (GremlingSS) [inline code/modules/research/designs/medical_designs.dm:650]: `| MECHFAB`

File: code/modules/research/designs/medical_designs.dm [hunk 11]
```diff
@@ -658,7 +657,7 @@
 	name = "Cybernetic Lungs"
 	desc = "A pair of cybernetic lungs."
 	id = "cybernetic_lungs"
-	build_type = MECHFAB
+	build_type = PROTOLATHE
```
Comment (GremlingSS) [inline code/modules/research/designs/medical_designs.dm:660]: `| MECHFAB`

File: code/modules/research/designs/medical_designs.dm [hunk 12]
```diff
@@ -668,7 +667,7 @@
 	name = "Upgraded Cybernetic Lungs"
 	desc = "A pair of upgraded cybernetic lungs."
 	id = "cybernetic_lungs_u"
-	build_type = MECHFAB
+	build_type = PROTOLATHE
```
Comment (GremlingSS) [inline code/modules/research/designs/medical_designs.dm:670]: `| MECHFAB`

File: code/modules/research/designs/medical_designs.dm [hunk 13]
```diff
@@ -678,7 +677,7 @@
 	name = "Cybernetic tongue"
 	desc = "A fancy cybernetic tongue."
 	id = "cybernetic_tongue"
-	build_type = MECHFAB
+	build_type = PROTOLATHE
```
Comment (GremlingSS) [inline code/modules/research/designs/medical_designs.dm:680]: `| MECHFAB`

File: code/modules/research/designs/stock_parts_designs.dm [hunk 14]
```diff
@@ -29,7 +29,7 @@
 	id = "basic_capacitor"
 	build_type = PROTOLATHE | AUTOLATHE
 	materials = list(/datum/material/iron = 100, /datum/material/glass = 100)
-	build_path = /obj/item/stock_parts/capacitor/simple
+	build_path = /obj/item/stock_parts/capacitor
```
Comment (GremlingSS) [inline code/modules/research/designs/stock_parts_designs.dm:32]: Revert this

File: code/modules/research/designs/stock_parts_designs.dm [hunk 15]
```diff
@@ -74,7 +74,7 @@
 	id = "basic_scanning"
 	build_type = PROTOLATHE | AUTOLATHE
 	materials = list(/datum/material/iron = 100, /datum/material/glass = 50)
-	build_path = /obj/item/stock_parts/scanning_module/simple
+	build_path = /obj/item/stock_parts/scanning_module
```
Comment (GremlingSS) [inline code/modules/research/designs/stock_parts_designs.dm:77]: This

File: code/modules/research/designs/stock_parts_designs.dm [hunk 16]
```diff
@@ -119,7 +119,7 @@
 	id = "micro_mani"
 	build_type = PROTOLATHE | AUTOLATHE
 	materials = list(/datum/material/iron = 100)
-	build_path = /obj/item/stock_parts/manipulator/simple
+	build_path = /obj/item/stock_parts/manipulator
```
Comment (GremlingSS) [inline code/modules/research/designs/stock_parts_designs.dm:122]: Dis

File: code/modules/research/designs/stock_parts_designs.dm [hunk 17]
```diff
@@ -164,7 +164,7 @@
 	id = "basic_micro_laser"
 	build_type = PROTOLATHE | AUTOLATHE
 	materials = list(/datum/material/iron = 100, /datum/material/glass = 50)
-	build_path = /obj/item/stock_parts/micro_laser/simple
+	build_path = /obj/item/stock_parts/micro_laser
```
Comment (GremlingSS) [inline code/modules/research/designs/stock_parts_designs.dm:167]: Pis

File: code/modules/research/designs/stock_parts_designs.dm [hunk 18]
```diff
@@ -208,7 +208,7 @@
 	id = "basic_matter_bin"
 	build_type = PROTOLATHE | AUTOLATHE
 	materials = list(/datum/material/iron = 100)
-	build_path = /obj/item/stock_parts/matter_bin/simple
+	build_path = /obj/item/stock_parts/matter_bin
```
Comment (GremlingSS) [inline code/modules/research/designs/stock_parts_designs.dm:211]: Piss.

**Review by GremlingSS — CHANGES_REQUESTED** [review]
Just these ~w~

File: code/game/objects/effects/step_triggers.dm (hunk 6 above)
Comment (FarmWizard) [reply code/game/objects/effects/step_triggers.dm:7]: Alright, donezo

File: code/game/objects/items/devices/portable_chem_blenderbelt.dm (hunk 7 above)
Comment (FarmWizard) [reply code/game/objects/items/devices/portable_chem_blenderbelt.dm:800]: And done

File: code/modules/research/designs/medical_designs.dm (hunk 11 above)
Comment (FarmWizard) [reply code/modules/research/designs/medical_designs.dm]: Done

File: code/modules/research/designs/medical_designs.dm (hunk 12 above)
Comment (FarmWizard) [reply code/modules/research/designs/medical_designs.dm]: Done

File: code/modules/research/designs/medical_designs.dm (hunk 13 above)
Comment (FarmWizard) [reply code/modules/research/designs/medical_designs.dm]: Done

File: code/modules/research/designs/medical_designs.dm (hunk 10 above)
Comment (FarmWizard) [reply code/modules/research/designs/medical_designs.dm]: Done

File: code/modules/research/designs/medical_designs.dm (hunk 9 above)
Comment (FarmWizard) [reply code/modules/research/designs/medical_designs.dm:630]: Finished

File: code/modules/research/designs/medical_designs.dm (hunk 8 above)
Comment (FarmWizard) [reply code/modules/research/designs/medical_designs.dm]: fixed

**Review by GremlingSS — APPROVED** [review]
I approve, looks good! 👍

**Comment by FarmWizard** [comment]
The only reason it's failed is because of step_trigger and I got told to revert adding var/client so it's good.

**Review by gozulio — APPROVED** [review]

**Pull request merged**